### ISSUE TITLE
Qubes Manager performance slowdown with 4.0.

### ISSUE THREAD

**Issue by nailgao** [QubesOS/qubes-issues#4116]
### Qubes OS version:

4.0.

### Affected component(s): 

Qubes Manager operations
all VMs

---

### Steps to reproduce the behavior:

move the pointer on a VM in Qubes Manager. Right-click and do something that affects the state of the VM (like start, kill, restart..)
and right after doing that, try to do something else in the Qubes Manager.

### Expected behavior:

The first selected operation on the selected VM gets started, and without delay you can move on to your second operation in Qubes Manager.

### Actual behavior:

After the first thing you do in Qubes Manager, this causes it to freeze for about 10 seconds or more, and only after that you can perform a second task in Qubes Manager.

### General notes:

with 4.0. the Qubes Manager performance has slowed down, and it spawned problems that did not exist with 3.2.

---

### Related issues:

some of the further Qubes Manager bugs so far witnessed are these:

-while fedora-27 VM has no apps selected in Qubes Manager, its drop-down menu shows "all" apps installed in fedora.
-a cloned VM does not inherit the selected apps of its source VM, unlike was the case in 3.2.

**Comment by andrewdavidwong** [comment]
Possibly subsumed by #3639.

**Comment by andrewdavidwong** [comment]
> some of the further Qubes Manager bugs so far witnessed are these:
> 
> -while fedora-27 VM has no apps selected in Qubes Manager, its drop-down menu shows "all" apps installed in fedora.
> -a cloned VM does not inherit the selected apps of its source VM, unlike was the case in 3.2.

Please file separate issues for these, if they haven't already been reported.

**Comment by nailgao** [comment]
@andrewdavidwong thank you, submitted in #4123 and #4124.

**Comment by Aekez** [comment]
@andrewdavidwong 
This bug doesn't seem isolated to the Qube Manager, something very similar also happens with the new Qubes 4.0. widget, furthermore it appears like a new bug (not much older than 1-2 weeks).
- A quick way to try reproduce, try use mouse to shutdown a series of VM's from the Qubes widget, after 2-3 VM's the widget start to freeze until the VM has fully shutdown.
- I think this relative new bug is somehow related to the fix that fixed the status updates in the Qubes widget, given the timing of the status update fix and this new freeze, is estimated to have happened around the same time.
- This bug was more aggressive in the second last current-testing updates (there abouts), and became less aggressive (though still a problem) in the newest current-testing updates. 
  - Since it's changing like this, Is it possibly a bug already aware of, and is being adjusted as we speak?

@nailgao can you try see if you can reproduce the same behavior in your Qubes widget, as you find in the Qube Manager?

**Comment by nailgao** [comment]
@Aekez I tried it with Qubes widget. It froze after having selected "shutdown" for 2-3 VMs, as you wrote.

this is not the only performance issue with Qubes Manager, nor with 4.0. I will add these as new issues, but I thought the following may be related and not needing a separate issue to be created for it, so I'm adding it here (if I am wrong please let me know, and I will create a separate issue):

- the started or killed VMs have these yellow and green dots. when you start or kill a VM, the dot is not shown until you first move away the highlight bar ( the blue bar) from the VM, and then highlight the VM back again. If you don't do this, you never see any dot concerning the status of a started or killed VM.

Creating a separate issue on usb widget does seem to be relevant, though. I will add the issue number here once I do it.

**Comment by nailgao** [comment]
as another Qubes Manager issue that I forgot to add to the last comment:
- relatively often and apparently as a novelty of 4.0. ( I have never met this alert on 3.2.), there comes a "could not connect within 60 seconds" error reporting that the start of a VM failed often during the occurence of the said slowness / freeze problem of Qubes Manager.
...possibly for the same reason, in several instances, the number of VMs allowed to be open at the same time have been 2 VMs less than before when I had 3.2.

**Comment by nailgao** [comment]
I keep adding Qubes Manager issues, please let me know if they require a separate title:

- when the "updates pending!" sign (the green download arrow) appeared next to a template VM, it used to be the case in 3.2. that when you right-click on "update VM", the respective VM used to be started first, and the update would automatically follow it.
however, with 4.0., when you select "update VM", the VM gets started, and then you need to select "update VM" again in order for the update to start.

**Comment by nailgao** [comment]
@andrewdavidwong  ok, sorry, you'd written this to me before but I just thought the latter issues might all be related to the same slowness mentioned in the title. 

For the issue added above concerning the green and yellow dots, I will create a new issue. 

For other appended issues, before posting them as new ones, I'd like to wait to see if they persist, for in 
the recent days I've run qubes template and dom0 updates, yet haven't had time with desktop applications long enough to observe them to see if the issues are perhaps  (hopefully) solved. (can it be the case? :))

**Comment by nailgao** [comment]
having just written about creating a new issue, in a second look, doesn't the problem with yellow-green dots seems rather intertwined with the main issue? 
The former is observed within the scope of the latter, hence the occurences are concurrent, so long as start/kill/restart VM is involved.

**Comment by andrewdavidwong** [comment]
> having just written about creating a new issue, in a second look, doesn't the problem with yellow-green dots seems rather intertwined with the main issue?
> The former is observed within the scope of the latter, hence the occurences are concurrent, so long as start/kill/restart VM is involved.

If you think it belongs in this issue, that's fine. Sometimes we don't know whether they're separate until the issue is diagnosed.
